.github/workflows/update_python_dependencies.yml

```diff
@@ -0,0 +1,38 @@
+name: Update Python Dependencies
+
+on:
+    workflow_dispatch:
+
+jobs:
+    update_dependencies:
+        runs-on: ubuntu-latest
+        steps:
+            - name: "Checkout"
+              uses: actions/checkout@v3
+              with:
+                  ref: main
+                  fetch-depth: 0
+            - name: Set up Python 3.8
+              id: py
+              uses: actions/setup-python@v2
+              with:
+                  python-version: 3.8
+                  cache: pip
+                  cache-dependency-path: "requirements/*.txt"
+            - name: Update dependencies
+              run: |
+                  pip install pip-tools
+                  make update-deps
+            - name: Commit changes and open PR
+              run: |
+                  export WEEK="w$(date +%V)"
+                  git config user.name github-actions
+                  git config user.email github-actions@github.com
+                  git switch -c "dependency-updates-$WEEK"
+                  git add requirements/*.txt 
+                  git commit -m "$WEEK dependency updates (automated)"
+                  git push origin "dependency-updates-$WEEK"
```

Comment:
💭 : This also _should_ work out of the box, but if it does have authentication issues we should switch to [`actions/checkout@v3`](https://github.com/actions/checkout#Push-a-commit-using-the-built-in-token).